sound/soc/sof/intel/hda.c

```diff
@@ -72,22 +72,38 @@ int hda_ctrl_dai_widget_setup(struct snd_soc_dapm_widget *w, bool setup)
 	if (setup) {
 		ret = sof_widget_setup(sdev, swidget);
 		if (ret < 0) {
-			dev_err(sdev->dev, "error: setting up DAI widget %s\n", w->name);
+			dev_err(sdev->dev, "error: failed setting up DAI widget %s\n", w->name);
 			return ret;
 		}
 
 		/* send DAI_CONFIG IPC */
-		return sof_ipc_tx_message(sdev->ipc, config->hdr.cmd, config, config->hdr.size,
+		ret = sof_ipc_tx_message(sdev->ipc, config->hdr.cmd, config, config->hdr.size,
 					  &reply, sizeof(reply));
+		if (ret < 0) {
+			dev_err(sdev->dev, "error: failed setting DAI config for %s\n", w->name);
+			return ret;
+		}
+
+		sof_dai->configured = true;
```

Comment:
It isn't a huge problem for me, but I think some layering purists would frown on this: a parameter is set to `true` at one level (in HDA) and to `false` at a different level, is there no way to do this all at the SOF core level? Not a huge deal for me, but if possible - that would look nicer